rosetta-2021.html

```diff
@@ -0,0 +1,236 @@
+<script>
+  // Rule of thumb: put a semicolon at the end of every line that doesn't end in a curly
+
+  // Simple output
+  // =============
+
+  // puts "howdy!"
+
+  // alert("howdy!");
+
+
+  // Declaring local variables
+  // =========================
+
+  // name = "Raghu"
+  // puts "howdy, #{name}!"
+
+  // var name = "Raghu";
+  // alert("howdy, " + name + "!");
+
+  // Note: there's no String interpolation in JS
+
+
+  // Defining and calling methods
+  // ============================
+
+  // def greet
+  //   puts "howdy!"
+  // end
+
+  // greet
+
+  // function greet() {
+  //   alert("howdy!");
+  // }
+
+  // greet();
+
+
+  // Methods that accept arguments
+  // =============================
+
+  // def greet(name)
+  //   puts "howdy, #{name}!"
+  // end
+
+  // function greet(name) {
+  //   alert("howdy, " + name + "!");
+  // }
+
+  // greet("Raghu");
+
+
+  // Conditionals
+  // ============
+
+  // city = "Toledo"
+  // if city == "Chicago"
+  //   puts "You live in the best city ever!"
+  // else
+  //   puts "You live in a so-so city."
+  // end
+
+  // var city = "Toledo";
+  // if (city === "Chicago") {
+  //   alert("You live in the best city ever!");
+  // }
+  // else {
+  //   alert("You live in a so-so city.");
+  // }
+
+  // Beware: there's also a == (non-strict equivalence) operator. Look it up.
+
+
+  // Incrementing integers
+  // =====================
+
+  // i += 1
+
+  // i++
```

Comment:
There's also `i--` for decrementing values, which I don't think exists in Ruby.

Reply:
In Ruby there isn't even an increment operator — Ruby's way cooler than that! Any operator of the pattern `x = x __ y` can be abbreviated `x __= y`. This is how our "or-equals" abbreviations when we memoize things works:

```ruby
def expensive_to_calculate_value
  @_expensive_to_calculate_value ||= expensive_calculation
  # @_expensive_to_calculate_value = @_expensive_to_calculate_value || expensive_calculation
end
```

Reply:
I was aware that you could shorten things like that in Ruby, I believe [it's similar in JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators#assignment_operators). I never looked too into why `i++`, or `i--` was never in Ruby since it's a part of most C-based languages.

Reply:
Huh, I guess then the question is why JS has both `i++` and `i += 1`/`i--` and `i -= 1` — just for familiarity/similarity with those other languages, I guess. In that light, sort of surprising that Ruby doesn't have `++` and `--`.